### PR TITLE
Finalize Tree View implementation.

### LINKED ISSUE #1222: Add `<sp-tree-view>` pattern
Add the Tree View pattern: https://opensource.adobe.com/spectrum-css/treeview.html

image

We may prioritize the flat vs nested versions for an initial release, but scoping is currently in progress for that.

### PR DESCRIPTION
**Pull request by Westbrook**
## Description
This has had a prerelease at https://www.npmjs.com/package/@spectrum-web-components/tree-view.

In that this has been pre-released already, I think it makes sense to get it into a more official "beta" situation and add it to the library for tighter scrutiny. If that makes sense as a path forward, I can update the Golden Images here so that CI should pass.

- add Tree View pattern, previously review at #1478
- add Tree View multi-select support #1423 
- add docs
  - https://westbrook-treeview--spectrum-web-components.netlify.app/components/tree-view/
  - https://westbrook-treeview--spectrum-web-components.netlify.app/components/tree-view-heading/
  - https://westbrook-treeview--spectrum-web-components.netlify.app/components/tree-view-item/
- add tests
- prep for release
- based on https://www.w3.org/TR/wai-aria-practices-1.2/#TreeView

## Related issue(s)
- fixes #1222

## Motivation and context
Spectrum coverage

## How has this been tested?
-   [ ] _Test case 1_
    1. Go [here](https://westbrook-treeview--spectrum-web-components.netlify.app/storybook/index.html?path=/story/tree-view--default)
    2. Experiment with the demos
    3. Compare their functionality to https://www.w3.org/TR/wai-aria-practices-1.2/#TreeView

## Types of changes
-   [x] New feature (non-breaking change which adds functionality)

## Checklist
-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

### CODE REVIEW

**Comment by sorinsarca** [comment]
A `details` slot (just after the label) will allow to add some details/actions to the item. Some examples include:

File size

tree-details-1

Favorites/Bookmarks

tree-details-2

Adding the content directly into the default slot (which is label) makes it almost
impossible to align, so a dedicated slot would be very helpful.

File: packages/tree-view/src/tree-view-item.css [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+:host([disabled]) #indicator {
+    pointer-events: none !important;
+}
+
+.link:before {
+    border-radius: var(--spectrum-treeview-item-border-radius);
+    border-width: var(--spectrum-treeview-item-border-size);
+}
+
```
Comment (sorinsarca) [inline packages/tree-view/src/tree-view-item.css:50]: Something like

```css
:host([selected]) .link:before {
  z-index: 1;
}
```

will prevent this ugly bug when hovering a sibling of a selected element

treeview-selected-bug

Comment (Westbrook) [reply packages/tree-view/src/tree-view-item.css:50]: There's actually a pretty sizable issue in the styling here https://github.com/adobe/spectrum-css/issues/1432 that likely needs upstream design guidance.

**Comment by Westbrook** [comment]
> A `details` slot (just after the label) will allow to add some details/actions to the item. Some examples include:

This seems like a good idea, but it not likely to be part of an initial shipment of this pattern as it is not represented [upstream in the Spectrum Design spec](https://spectrum.adobe.com/page/tree-view/). We'll raise this with their team and see whether they are interested in pursuing this.

**Comment by sorinsarca** [comment]
I see that there is now an optional [Actions area](https://spectrum.adobe.com/page/tree-view/#Actions-area) to which I referred as "details" slot. I don't know if it was there from the beginning.

**Comment by Westbrook** [comment]
@sorinsarca the upstream spec changing a good amount is one of many reasons work here has paused. It’s not likely we will return to this until late August to September when there is updated input from the design and CSS teams.

**Comment by Westbrook** [comment]
Will reopen with upcoming CSS changes.

**Comment by ranjeet-choudhary** [comment]
@Westbrook any plans to reopen if the above issues are resolved?

**Comment by najikahalsema** [comment]
@ranjeet-choudhary Hi! CSS team has migrated this component to Core Tokens, but the previous issues still need to be addressed as of 23 days ago. I just followed up with @pfulton on the status of this.

**Comment by pfulton** [comment]
Thanks @najikahalsema! We r[eleased](https://github.com/adobe/spectrum-css/commit/8b13743cc20786f5a71fbbc22714c9deadb3febb) several fixes for our implementation of Tree View today.

You can view the latest docs [here](https://opensource.adobe.com/spectrum-css/treeview.html) and also see the component in action in our [Storybook](https://opensource.adobe.com/spectrum-css/preview/?path=/story/components-tree-view--default).

**Pull request closed without merging**